internal/pkg/agent/application/apm_config_modifier.go

```diff
@@ -0,0 +1,146 @@
+// Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+// or more contributor license agreements. Licensed under the Elastic License;
+// you may not use this file except in compliance with the Elastic License.
+
+package application
+
+import (
+	"fmt"
+
+	"github.com/elastic/elastic-agent-client/v7/pkg/proto"
+
+	"github.com/elastic/elastic-agent/internal/pkg/agent/application/coordinator"
+	"github.com/elastic/elastic-agent/internal/pkg/config"
+	monitoringcfg "github.com/elastic/elastic-agent/internal/pkg/core/monitoring/config"
+	"github.com/elastic/elastic-agent/pkg/component/runtime"
+	"github.com/elastic/elastic-agent/pkg/core/logger"
+
+	"github.com/elastic/elastic-agent/internal/pkg/agent/errors"
+	"github.com/elastic/elastic-agent/pkg/component"
+	"github.com/elastic/elastic-agent/pkg/utils"
+)
+
+func InjectAPMConfig(comps []component.Component, cfg map[string]interface{}) ([]component.Component, error) {
```

Comment:
As someone who doesn't know much/anything about elastic-agent, I find it hard to understand why we have to inject and then patch. How about documenting those two public methods?

Reply:
I will add some documentation, the short version is:
- `PatchAPMConfig` will patch agent configuration coming from fleet (which does not support APM configuration at the moment) by adding the APM config values retrieved from the elastic-agent.yaml file
- `InjectAPMConfig` will add APM configuration to the internal datamodel agent uses to determine which components and units to run